lib/features.js

```diff
@@ -0,0 +1,67 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+module.exports = config => {
+  const lastAccessTimeUpdates = config.lastAccessTimeUpdates
+  const signinConfirmation = config.signinConfirmation
+
+  return {
+    isSampledUser: isSampledUser,
+
+    isLastAccessTimeEnabled (uid, email) {
+      return lastAccessTimeUpdates.enabled && (
+        isSampledUser(uid, lastAccessTimeUpdates.sampleRate) ||
+        new RegExp(lastAccessTimeUpdates.enabledEmailAddresses).test(email)
+      )
+    },
+
+    isSigninConfirmationEnabled (uid, email, request) {
+      if (! signinConfirmation.enabled) {
+        return false
+      }
+
+      // Always create unverified tokens if customs-server
+      // has said the request is suspicious.
+      if (request.app.isSuspiciousRequest) {
+        return true
+      }
+
+      // Or if the email address matches one of these regexes.
+      if (signinConfirmation.forceEmailRegex.some(regex => new RegExp(regex).test(email))) {
+        return true
+      }
+
+      // While we're testing this feature, there may be some funky
+      // edge-cases in device login flows that haven't been fully tested.
+      // Temporarily avoid them for regular users by checking the `context` flag,
+      // and create pre-verified sessions for unsupported clients.
+      // This check will go away in the final version of this feature.
+      const context = request.payload &&
+        request.payload.metricsContext &&
+        request.payload.metricsContext.context
+      if (signinConfirmation.supportedClients.indexOf(context) === -1) {
+        return false
+      }
+
+      // Check to see if user in roll-out cohort.
+      return isSampledUser(uid, signinConfirmation.sample_rate)
+    }
+  }
+}
+
+function isSampledUser (uid, sampleRate) {
```

Comment:
The trick here, is that if you have both features enabled at 10%, the *same* 10% of users are going to be opted in to both features.  That has bad implications for doing any statistical analysis of the results.

The typical way to decouple different experiments, is to hash together the name of the experiment with the uid before interpreting it as a number.  Like this:

  https://github.com/dannycoates/abatar/blob/master/experiment.js#L140